### ISSUE TITLE
#6.3.6 and 6.3.7 weird loss values during the training

### ISSUE THREAD

**Issue by Juusotak** [fchollet/deep-learning-with-python-notebooks#127]
I executed the codes in sections #6.3.7 and #6.3.6 in my PC. The loss values during the training and validation seem to differ vastly from the values presented here in GitHub or in the book. As I fit the recurrent network using dropout in the section 6.3.6, the loss value of the first epoch was for example 80528032321365409792.0000. In the section 6.3.7 keras only displayed "nan" as training and validation loss value.
Is somebody facing the same issues? Any suggestions what is the cause of this issue?

**Comment by the-drunk-coder** [comment]
Hmm I'm also having weird results, on the example 6.3.4, "A basic machine learning approach", running the notebook on google colab.

Training takes significantly longer, and the loss values are constantly rising. 

![image](https://user-images.githubusercontent.com/9823721/73376468-c9327200-42bd-11ea-8247-f6e4a504a11a.png)

**Comment by the-drunk-coder** [comment]
Now I switched the Colab runtime type from "None" to "GPU", now the results look like this: 
![image](https://user-images.githubusercontent.com/9823721/73377193-f16ea080-42be-11ea-96b7-6d65fabf4813.png)
Better, but still very different from the book.

@Juusotak are you using a GPU ?

**Comment by Juusotak** [comment]
@the-drunk-coder yes I'm using the GPU version of the Keras.  The loss and val_loss values are as follows thru out the whole fitting process. 

![Annotation 2020-01-29 213047 PNG](https://user-images.githubusercontent.com/37575800/73391058-f09a3680-42df-11ea-90d4-bf1663349991.jpg)

I'm running the code on Spyder 3.6 and I assume the GPU computation is working properly since I can find my GPU at the following output.

![Annotation 2020-01-29 214700](https://user-images.githubusercontent.com/37575800/73391610-f5abb580-42e0-11ea-95f0-c6047ca0d245.jpg)

**Comment by nickeduncan** [comment]
Also having the same issue as @Juusotak, my graph for listing 6.38 looks the same as @the-drunk-coder, and then my figure for listing 6.39 look like this:

**Comment by hanyun2019** [comment]
Hi @nickeduncan,

I have found the same problem with you:
"Listing 6.41 has loss and val_loss values returning NaN"

Have you found the solution for this problem? Thanks.

Best,
Haowen

**Comment by chriswininger** [comment]
yeah sadly, been seeing the same with this example, sometimes really high loss values, other times NAN

![Screenshot from 2021-07-23 06-57-15](https://user-images.githubusercontent.com/1688763/126772659-2970ccef-2607-4820-beed-04e041eaab88.png)

All the other examples have worked flawlessly, not sure what's up with this one

**Comment by fangxuehouwuming** [comment]
> I executed the codes in sections #6.3.7 and #6.3.6 in my PC. The loss values during the training and validation seem to differ vastly from the values presented here in GitHub or in the book. As I fit the recurrent network using dropout in the section 6.3.6, the loss value of the first epoch was for example 80528032321365409792.0000. In the section 6.3.7 keras only displayed "nan" as training and validation loss value. Is somebody facing the same issues? Any suggestions what is the cause of this issue?

Hi, I met this problem too. Have you solved it ?
